packages/nouns-contracts/tasks/populate-descriptor.ts

```diff
@@ -16,32 +18,75 @@ task('populate-descriptor', 'Populates the descriptor with color palettes and No
     types.string,
   )
   .setAction(async ({ nftDescriptor, nounsDescriptor }, { ethers }) => {
-    const descriptorFactory = await ethers.getContractFactory('NounsDescriptor', {
+    const descriptorFactory = await ethers.getContractFactory('NounsDescriptorV2', {
       libraries: {
-        NFTDescriptor: nftDescriptor,
+        NFTDescriptorV2: nftDescriptor,
       },
     });
     const descriptorContract = descriptorFactory.attach(nounsDescriptor);
 
     const { bgcolors, palette, images } = ImageData;
     const { bodies, accessories, heads, glasses } = images;
 
-    // Chunk head and accessory population due to high gas usage
-    await descriptorContract.addManyBackgrounds(bgcolors);
-    await descriptorContract.addManyColorsToPalette(0, palette);
-    await descriptorContract.addManyBodies(bodies.map(({ data }) => data));
-
-    const accessoryChunk = chunkArray(accessories, 10);
-    for (const chunk of accessoryChunk) {
-      await descriptorContract.addManyAccessories(chunk.map(({ data }) => data));
-    }
+    const {
+      encodedCompressed: bodiesCompressed,
+      originalLength: bodiesLength,
+      itemCount: bodiesCount,
+    } = dataToDescriptorInput(bodies.map(({ data }) => data));
+    const {
+      encodedCompressed: accessoriesCompressed,
+      originalLength: accessoriesLength,
+      itemCount: accessoriesCount,
+    } = dataToDescriptorInput(accessories.map(({ data }) => data));
+    const {
+      encodedCompressed: headsCompressed,
+      originalLength: headsLength,
+      itemCount: headsCount,
+    } = dataToDescriptorInput(heads.map(({ data }) => data));
+    const {
+      encodedCompressed: glassesCompressed,
+      originalLength: glassesLength,
+      itemCount: glassesCount,
+    } = dataToDescriptorInput(glasses.map(({ data }) => data));
 
-    const headChunk = chunkArray(heads, 10);
-    for (const chunk of headChunk) {
-      await descriptorContract.addManyHeads(chunk.map(({ data }) => data));
-    }
+    await descriptorContract.addManyBackgrounds(bgcolors);
+    await descriptorContract.setPalette(0, `0x000000${palette.join('')}`);
 
-    await descriptorContract.addManyGlasses(glasses.map(({ data }) => data));
+    await descriptorContract.addBodies(bodiesCompressed, bodiesLength, bodiesCount, {
+      gasLimit: 30_000_000,
```

Comment:
@eladmallel why is specifying the gas limit needed?

Reply:
when I ran the script without it I got this error
```
InvalidInputError: Transaction gas limit is 30488472 and exceeds block gas limit of 30000000
```

I tried setting the `gas` network property in `hardhat.config.ts` but that didn't fix the issue.

happy to learn and do something better!